File: alpha_engine/alerts.py
import json
import logging
from urllib import parse, request
from .config import settings

log = logging.getLogger(__name__)


def notify(message: str) -> bool:
    """Send a Telegram notification when credentials are configured."""
    if not settings.telegram_bot_token or not settings.telegram_chat_id:
        return False
    url = f"https://api.telegram.org/bot{settings.telegram_bot_token}/sendMessage"
    payload = parse.urlencode({
        "chat_id": settings.telegram_chat_id,
        "text": message[:4000],
        "disable_web_page_preview": "true",
    }).encode()
    try:
        req = request.Request(url, data=payload, method="POST")
        with request.urlopen(req, timeout=10) as response:
            body = json.loads(response.read().decode("utf-8"))
            return bool(body.get("ok"))
    except Exception:
        log.exception("Telegram notification failed")
        return False

File: worker/src/alerts.ts
import type { Settings } from "./config";

export async function notify(settings: Settings, message: string): Promise<boolean> {
  if (!settings.telegramBotToken || !settings.telegramChatId) return false;
  const url = `https://api.telegram.org/bot${settings.telegramBotToken}/sendMessage`;
  try {
    const response = await fetch(url, {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: new URLSearchParams({
        chat_id: settings.telegramChatId,
        text: message.slice(0, 4000),
        disable_web_page_preview: "true",
      }),
      signal: AbortSignal.timeout(10000),
    });
    const body = (await response.json()) as { ok?: boolean };
    return Boolean(body.ok);
  } catch {
    return false;
  }
}
